crates/ide-assists/src/handlers/extract_function.rs

```diff
@@ -70,6 +70,11 @@ pub(crate) fn extract_function(acc: &mut Assists, ctx: &AssistContext<'_>) -> Op
     }
 
     let node = ctx.covering_element();
+    if node.kind() == SyntaxKind::L_CURLY || node.kind() == SyntaxKind::R_CURLY {
```

Comment:
While we are at this, let's expand this to all kinds of braces (for brevity we can also use the `T!` macro, so `T!['{']`, `T!['(']`, ... which expand to the appropriate `SyntaxKind`)

Reply:
Done i expanded it to support (, [, {